pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -523,21 +523,46 @@ type EC2SDConfig struct {
 	// +optional
 	SecretKey *corev1.SecretKeySelector `json:"secretKey,omitempty"`
 	// AWS Role ARN, an alternative to using AWS API keys.
+	// +kubebuilder:validation:MinLength=1
 	// +optional
 	RoleARN *string `json:"roleARN,omitempty"`
-	// RefreshInterval configures the refresh interval at which Prometheus will re-read the instance list.
-	// +optional
-	RefreshInterval *v1.Duration `json:"refreshInterval,omitempty"`
 	// The port to scrape metrics from. If using the public IP address, this must
 	// instead be specified in the relabeling rule.
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=65535
 	// +optional
 	Port *int `json:"port"`
+	// RefreshInterval configures the refresh interval at which Prometheus will re-read the instance list.
+	// +optional
+	RefreshInterval *v1.Duration `json:"refreshInterval,omitempty"`
 	// Filters can be used optionally to filter the instance list by other criteria.
 	// Available filter criteria can be found here:
 	// https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeInstances.html
 	// Filter API documentation: https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_Filter.html
 	// +optional
 	Filters Filters `json:"filters,omitempty"`
+	// Optional HTTP basic authentication information.
+	// Cannot be set at the same time as `authorization`, or `oauth2`.
+	// +optional
+	BasicAuth *v1.BasicAuth `json:"basicAuth,omitempty"`
```

Comment:
Looking at https://prometheus.io/docs/prometheus/latest/configuration/configuration/#ec2_sd_config
I don't see BasicAuth, Authorization, OAuth2 is not yet supported? can you verify

Reply:
Not supported by AWS yet the reason why we didn't add it so I don't think there would be a use adding it

Reply:
Hmm... I am not sure of that, I see it in Prometheus code as well as the doc link you sent. Could you confirm that it is not supported by AWS?

```
# Optional HTTP basic authentication information, currently not supported by AWS.
basic_auth:
  [ username: [<string>](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#string) ]
  [ password: [<secret>](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#secret) ]
  [ password_file: [<string>](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#string) ]

# Optional `Authorization` header configuration, currently not supported by AWS.
authorization:
  # Sets the authentication type.
  [ type: [<string>](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#string) | default: Bearer ]
  # Sets the credentials. It is mutually exclusive with
  # `credentials_file`.
  [ credentials: [<secret>](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#secret) ]
  # Sets the credentials to the credentials read from the configured file.
  # It is mutuall exclusive with `credentials`.
  [ credentials_file: [<filename>](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#filename) ]

# Optional OAuth 2.0 configuration, currently not supported by AWS.
oauth2:
  [ [<oauth2>](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#oauth2) ]

# Optional proxy URL.
[ proxy_url: [<string>](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#string) ]
# Comma-separated string that can contain IPs, CIDR notation, domain names
# that should be excluded from proxying. IP and domain names can
# contain port numbers.
[ no_proxy: [<string>](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#string) ]
# Use proxy URL indicated by environment variables (HTTP_PROXY, https_proxy, HTTPs_PROXY, https_proxy, and no_proxy)
[ proxy_from_environment: [<boolean>](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#boolean) | default: false ]
# Specifies headers to send to proxies during CONNECT requests.
[ proxy_connect_header:
  [ [<string>](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#string): [[<secret>](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#secret), ...] ] ]
```

Reply:
I see, I just saw it as a part of Prometheus code and decided to add it as well.

Reply:
Should I remove it?

Reply:
IMHO we should add only fields that are useful and supported. But I don't have strong opinion on this. We can also add fields later based on requests.

But we can see what @simonpasquier  thinks about this.

Reply:
Ok, how about we remove it now and add it later if we think it is necessary? As you pointed out, doesn't make much sense to add fields that are not supported. Though I don't understand why they are part of Prometheus in the first place.

Reply:
Hmm BasicAuth, Authorization and OAuth2 are supported by Prometheus for EC2 SD. I see no reason to not support them here too (though it can be a follow-up PR).

Reply:
https://prometheus.io/docs/prometheus/latest/configuration/configuration/#ec2_sd_config states its not supported by aws. IIUC its added common for all SDs through http client config

Reply:
Ah right! The information is bit buried in the text. I agree then that we should remove the fields from the CRD.

Reply:
I also bet that the same is true for many other SDs...

Reply:
IIRC for many SDs we haven't added I am sure for the ones I added but would be good to review

Reply:
This will be a bit tedious but doing it one at a time should not be too much of a problem. Let me also check the ones we have already refactored.